homeassistant/components/rainmachine/__init__.py

```diff
@@ -313,9 +313,10 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if unload_ok:
         hass.data[DOMAIN].pop(entry.entry_id)
 
-    if len(hass.config_entries.async_entries(DOMAIN)) == 1:
-        # If this is the last instance of RainMachine, deregister any services defined
-        # during integration setup:
+    entries = hass.config_entries.async_entries(DOMAIN)
+    if len(entries) == 1 and entries[0].state == ConfigEntryState.LOADED:
```

Comment:
The unload function that we're inside will only be called if the config entry is loaded. I think the comment meant that we should count the loaded entries only and check that it's one. If it is one, we know that it's this one.

Reply:
Got it!